kegeland-api/src/wheelchairPatients/wheelchairPatients.service.ts

```diff
@@ -97,5 +99,101 @@ export class WheelchairPatientsService {
       });
     return { id, ...data };
   }
+
+  /**
+   * Function for adding a new empty game session to a specific wheelchair patient
+   * @param id ID of the wheelchair patient
+   * @returns The ID of the newly created game session
+   */
+  async addEmptyGameSessionToPatient(id: string) {
+    try {
+      const gameSessionsCollection = this.firebaseService.firestore
+        .collection('patients')
+        .doc(id)
+        .collection('gameSessions');
+
+      const newSession = {
+        createdAt: firestore.Timestamp.now(),
+      };
+
+      const newSessionRef = await gameSessionsCollection.add(newSession);
+
+      return { id: newSessionRef.id };
+    } catch (error) {
+      throw new InternalServerErrorException(error.message);
+    }
+  }
+
+  /**
+   * Function for updating a game session of a specific wheelchair patient
+   * @param patientId ID of the wheelchair patient
+   * @param sessionId ID of the game session to update
+   * @param gameSession Data to update the game session with
+   */
+  async updateGameSession(patientId: string, sessionId: string, gameSession: UpdateGameSessionDto) {
+    console.log(gameSession.startTime);
+    console.log("test");
+    try {
+      // Verify that startTime and endTime are valid Date instances
+      if (!(gameSession.startTime instanceof Date)) {
+          throw new Error("startTime is not a valid Date object");
+      }
+      if (!(gameSession.endTime instanceof Date)) {
+          throw new Error("endTime is not a valid Date object");
+      }
+      
+      const gameSessionDoc = this.firebaseService.firestore
+        .collection('patients')
+        .doc(patientId)
+        .collection('gameSessions')
+        .doc(sessionId);
+
+      // Check if the document exists
+      const docSnapshot = await gameSessionDoc.get();
```

Comment:
The following lines holds a lot of duplicate code. Possible to create a helperfunction that both checks if value is instance of date, and transforms it to firebase time object?